File: backend_analisis_exergetico/schemas/flujo.py
from fastapi_utils.api_model import APIModel

from schemas.sustancia_en_flujo import SustanciaEnFlujo


class FlujoBase(APIModel):
    pass


class FlujoCreate(FlujoBase):
    pass


class FlujoUpdate(FlujoBase):
    pass


class Flujo(FlujoBase):
    id: int
    sustancias: list[SustanciaEnFlujo]

File: frontend_analisis_exergetico/src/modules/exergia/models/flujo.ts
import { BaseModel } from 'src/utils/types';
import { SustanciaEnFlujo } from './sustancia-en-flujo';

export type FlujoBase = BaseModel;

export type FlujoCreateBase = FlujoBase;
export type FlujoUpdateBase = FlujoBase;

export interface Flujo extends FlujoBase {
  id: number;
  sustancias: SustanciaEnFlujo[];
}

export type TipoFlujo = 'Entrada' | 'Salida';
